File: #0746.min-cost-climbing-stairs.cpp
class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {

        int size = cost.size();
        vector<int> v = vector<int>(size + 2, 0);
        for (int i = size - 1; i >= 0; i--) {
            v[i] = cost[i] + std::min(v[i + 1], v[i + 2]);
        }
        return std::min(v[0], v[1]);
    }
};

File: #0746.min-cost-climbing-stairs.js
/**
 * @param {number[]} cost
 * @return {number}
 */
var minCostClimbingStairs = function(cost) {
    let arr = cost.slice(0, 2);
    for (let i = 2; i < cost.length; i++) {
        [arr[0], arr[1]] = [arr[1], arr[0]];
        arr[1] = Math.min(arr[0], arr[1]) + cost[i];
    }
    return arr[0] > arr[1] ? arr[1] : arr[0];
};
